Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -1161,6 +1161,13 @@ static NSTrackingAreaOptions trackingAreaOptions()
     if (m_page->preferences().siteIsolationEnabled())
         useRemoteLayerTree = true;
 
+    // Temporarily disable UI side compositing in Recovery OS.
+    // FIXME: remove this workaround once a proper fix is in place.
+    if (!getuid()) {
```

Comment:
Let's wrap this in a helper function called `isInRecoveryOS` or something so it’s less mysterious. If we have existing code for this, share it.

Reply:
Done!

Thanks for reviewing :)